disnake/state.py

```diff
@@ -777,12 +778,19 @@ def parse_message_create(self, data: gateway.MessageCreateEvent) -> None:
         # we ensure that the channel is a type that implements last_message_id
         if channel and channel.__class__ in (TextChannel, Thread, VoiceChannel):
             channel.last_message_id = message.id  # type: ignore
+        if channel and isinstance(channel, Thread):
+            channel.total_message_sent += 1
+            channel.message_count += 1
```

Comment:
This should check `message.type is MessageType.thread_starter_message` and not increment the counter in that case, to mirror the API behavior:
(lib incremented counters, API didn't)
![image](https://user-images.githubusercontent.com/8530778/188516104-408232b3-0a6a-4471-b4e1-28f7d85d3771.png)

Reply:
> This should check `message.type is MessageType.thread_starter_message` and not increment the counter in that case, to mirror the API behavior: (lib incremented counters, API didn't) ![image](https://user-images.githubusercontent.com/8530778/188516104-408232b3-0a6a-4471-b4e1-28f7d85d3771.png)

`Thread.total_message_sent` should count the starter message so i've applied the changes only to `Thread.message_count`, lemme know if that's right

Reply:
I wasn't sure about that part, but it doesn't seem like that's the case - `total_message_sent` also shouldn't count the starter message:
<img src="https://user-images.githubusercontent.com/8530778/188649135-651f5559-235f-4dc9-9815-fefb921a2177.png" width="600"/>

<details>
<summary>code</summary>

```py
# create thread from message
msg = await ctx.send("test")
t_created = await msg.create_thread(name="h")
t_id = t_created.id

# check values in cache
t_cache = ctx.guild.get_thread(t_id)
print("cache:", t_cache.message_count, t_cache.total_message_sent)

# check values in api
t_fetch = await ctx.guild.fetch_channel(t_id)
print("fetch:", t_fetch.message_count, t_fetch.total_message_sent)
```

</details>